src/components/directory-item-selector.js

```diff
@@ -0,0 +1,100 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import TreeViewFinder from '@gridsuite/commons-ui/lib/components/TreeViewFinder';
+import PropTypes from 'prop-types';
+import { fetchDirectoryContent, fetchRootFolders } from '../utils/rest-api';
+import { makeStyles } from '@material-ui/core/styles';
+import {
+    getIconFor,
+    elementType,
+} from '@gridsuite/commons-ui/lib/utils/elementType';
+
+const useStyles = makeStyles((theme) => ({
+    icon: {
+        marginRight: theme.spacing(1),
+        width: '18px',
+        height: '18px',
+    },
+}));
+
+const DirectoryItemSelector = (props) => {
+    const [data, setData] = useState([]);
+    const nodeMap = useRef({});
```

Comment:
why do you need this nodeMap, you have the data state which is made for this usage.